transaction-service/src/main/java/pl/adamboguszewski/transaction/service/infrastructure/exception/TransactionExceptionHandler.java

```diff
@@ -29,10 +32,29 @@
     @ResponseStatus(HttpStatus.NO_CONTENT)
     @ExceptionHandler(IllegalCurrencyArgumentException.class)
     public ResponseEntity<CreateTransactionResponse> handle(IllegalCurrencyArgumentException exception) {
-        log.info(exception.getMessage());
+        log.info("Currency " + exception.getCurrency() + " not recognized.");
+        //[TODO] Handle error code
+        return new ResponseEntity<>(
+                new CreateTransactionFailureResponse(exception.getMessage(), 2137L),
+                HttpStatus.NO_CONTENT);
+    }
+
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @ExceptionHandler(IllegalPaymentTypeArgumentException.class)
+    public ResponseEntity<CreateTransactionResponse> handle(IllegalPaymentTypeArgumentException exception) {
+        log.info("Payment type " + exception.getPaymentType() + " not recognized.");
         //[TODO] Handle error code
         return new ResponseEntity<>(
                 new CreateTransactionFailureResponse(exception.getMessage(), 2137L),
                 HttpStatus.NO_CONTENT);
     }
+
+    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
+    @ExceptionHandler(Exception.class)
+    public ResponseEntity<Void> handleUnexpectedExceptions(Exception exception) {
```

Comment:
Now user know only that something fucked up. At least, AT LEAST return handled exception